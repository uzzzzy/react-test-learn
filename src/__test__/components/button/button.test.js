import React from 'react'
import renderer from 'react-test-renderer'

import Button from '../../../components/button/button'

import { render, screen, cleanup } from '@testing-library/react'

import '@testing-library/jest-dom'

afterEach(cleanup)
it('render without crashing', () => {
    render(<Button></Button>)
})

it('renders button correctly', () => {
    render(<Button label="Click Please!"></Button>)

    expect(screen.getByTestId('button')).toHaveTextContent('Click Please!')
})
it('renders button test', () => {
    render(<Button label="test"></Button>)

    expect(screen.getByTestId('button')).toHaveTextContent('test')
})
it('renders button help', () => {
    render(<Button label="help"></Button>)

    expect(screen.getByTestId('button')).toHaveTextContent('help')
})

it('matches snapshot', () => {
    const tree = renderer.create(<Button label="save" />).toJSON()

    expect(tree).toMatchSnapshot()
})
