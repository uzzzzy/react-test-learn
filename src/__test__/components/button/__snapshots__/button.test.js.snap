// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="rounded border px-3 py-1 border-teal-300 font-bold text-teal-300 cursor-pointer hover:bg-teal-300 hover:text-slate-800 hover:shadow-teal-300 hover:shadow"
  data-testid="button"
>
  Hi, 
  save
</div>
`;
